src/main/java/io/deeplay/Board.java

```
@@ -1,43 +1,77 @@
+            board[center][center - 1] = Cell.BLACK;
+        }
     }
 
-    public void setOnPlace(int i, int j, Disk disk) { //установка диска в ячейку
-        board[i][j] = disk;
+
+    /**
+     * установка диска в указанную ячейку
+     */
+    public void setOnPlace(int row, int col, Cell cell) {
+        try {
+            board[row][col] = cell;
+        } catch (IllegalArgumentException exception) {
+            System.out.println("неверные координаты, повторите ввод");
+        }
+
     }
 
-    public Disk whatIsInside(int i, int j) { //проверка(возврат), что в ячейке
-        return board[i][j];
+    /**
+     * возвращает значение cell, которое лежит в клетке
+     */
+    public Cell getFromPlace(int row, int col) {
+        try {
+            return board[row][col];
+        } catch (IllegalArgumentException exception) {
+            System.out.println("неверные координаты, повторите ввод");
+            return null;
+        }
     }
 
-    public int countDisk(Disk disk) { //подсчет дисков указанного цвета
+    /**
+     * считает диски указанного цвета
+     */
+    public int countDisk(Cell cell) {
         int count = 0;
-        for (Disk[] disks : board) {
-            for (int j = 0; j < deskSize; j++) {
-                if (disks[j] == disk) {
+        for (Cell[] cells : board) {
+            for (int col = 0; col < boardSize; col++) {
+                if (cells[col] == cell) {
                     count++;
                 }
             }
         }
         return count;
     }
 
-    private boolean isValidMove(int row, int col, Player player) { //проверка, является ли клетка подходящей для хода указанного игрока(здесь ли реализовывать?)
-        if (row < 0 || row >= deskSize || col < 0 || col >= deskSize || board[row][col] != null) {
+    private boolean isValidMove(int row, int col, Player player) {
+        if (row < 0 || row >= boardSize || col < 0 || col >= boardSize || board[row][col] != null) {
```

1.  Мы поменяли enum Cell, теперь у нас в массиве не может быть null. 
2. Проверку вынести в отдельный метод, который будет бросать исключение при неправильных индексах. Его и использовать во всех методах для проверки.
3. Возможно, если мы сюда неправильные индексы передали, то можно оценивать ход как неправильный. НО! это ошибка в ключевой логике игры, которой здесь не должно быть. Мы должны в этом месте "упасть", выдать исключение с подробным сообщением о том, что пошло не так. А потом исправить эту ошибку. Стратегия Fail fast.

---

Александр этим займется